src/pspm_sf.m

```diff
@@ -84,135 +84,81 @@
     varargout{2} = outfile;
 end
 %% 2 Check input
-% 2.1 Check missing input --
-if nargin<1
-  warning('ID:invalid_input', 'Nothing to do.'); return;
-elseif nargin<2
-  options = struct();
-end
-if ~isfield(model, 'datafile')
-  warning('ID:invalid_input', 'No input data file specified.'); return;
-elseif ~isfield(model, 'modelfile')
-  warning('ID:invalid_input', 'No output model file specified.'); return;
-elseif ~isfield(model, 'timeunits')
-  warning('ID:invalid_input', 'No timeunits specified.'); return;
-elseif ~isfield(model, 'timing') && ~strcmpi(model.timeunits, 'file')
-  warning('ID:invalid_input', 'No epochs specified.'); return;
-end
-% 2.2 Check faulty input --
-if ~ischar(model.datafile) && ~iscell(model.datafile)
-  warning('ID:invalid_input', 'Input data must be a cell or string.'); return;
-elseif ~ischar(model.modelfile) && ~iscell(model.modelfile)
-  warning('ID:invalid_input', 'Output model must be a string.'); return;
-elseif ~ischar(model.timing) && ~iscell(model.timing) && ~isnumeric(model.timing)
-  warning('ID:invalid_input', 'Event onsets must be a string, cell, or struct.'); return;
-elseif ~ischar(model.timeunits) || ~ismember(model.timeunits, {'seconds', 'markers', 'samples', 'whole'})
-  warning('ID:invalid_input',...
-    'Timeunits (%s) not recognised; ',...
-    'only ''seconds'', ''markers'', ''samples'' and ''whole'' are supported',...
-    model.timeunits); return;
-end
-% 2.3 Convert single file input to cell --
-if ischar(model.datafile)
-  model.datafile={model.datafile};
-end
-if ischar(model.timing) || isnumeric(model.timing)
-  model.timing = {model.timing};
+% 2.1 check missing input --
+if nargin < 1; errmsg = 'Nothing to do.'; warning('ID:invalid_input', errmsg); return
+elseif nargin < 2; options = struct(); end
+
+% 2.2 check model
+model = pspm_check_model(model, 'glm');
+if model.invalid
+    return
 end
-if ischar(model.modelfile)
-  model.modelfile = {model.modelfile};
+
+% 2.3 check options 
+options = pspm_options(options, 'glm');
```

Comment:
@dominikbach May I ask why this is using `glm`'s options and model settings? This is `sf` and I originally write the options for `sf` in `pspm_options`...